app/src/main/java/org/jellyfin/mobile/fragment/WebViewFragment.kt

```diff
@@ -185,7 +182,27 @@ class WebViewFragment : Fragment(), NativePlayerHost {
                 if (request.url.toString() == instanceUrl) onErrorReceived()
             }
         }
-        webChromeClient = WebChromeClient()
+        webChromeClient = object : WebChromeClient() {
+            override fun onConsoleMessage(consoleMessage: ConsoleMessage?): Boolean {
+                if (consoleMessage?.messageLevel() == null) {
+                    // Use default implementation if null
+                    return super.onConsoleMessage(consoleMessage)
+                }
+
+                val formattedConsoleMessage = "${consoleMessage.message()}, ${consoleMessage.sourceId()} (${consoleMessage.lineNumber()})"
+
+                Timber.tag("WebView")
+                when(consoleMessage.messageLevel()) {
+                    ConsoleMessage.MessageLevel.ERROR -> Timber.e(formattedConsoleMessage)
+                    ConsoleMessage.MessageLevel.WARNING -> Timber.w(formattedConsoleMessage)
+                    ConsoleMessage.MessageLevel.DEBUG -> Timber.d(formattedConsoleMessage)
+                    ConsoleMessage.MessageLevel.TIP -> Timber.v(formattedConsoleMessage)
+                    else -> Timber.i(formattedConsoleMessage)
+                }
```

Comment:
Why not map the ConsoleMessage.MessageLevel entries to Log entries and then use Timber.log()?